Add fixed mode on paper side

ialokim opened the pull request:
As discussed in #4, this PR adds a fixed mode neither gates can be moved nor wires added. The mode can be set by `paper.fixed(boolean)`. You may try it using the "fixed mode" checkbox in the `template.html`.

Currently, the CSS rules feel kind of hacky as I still want to respond to mouseovers on gates where the displayed base (hex, dec etc.) may be changed or on subcircuits and memory to be able to open them in a popup window. Also clicks on inputs should be reacted on. I'm not sure if I even found all of the cases. Perhaps we could add some extra css class to the elements that still should respond to mouse events s.th. the rules would look nicer.

Also fixed mode is still not applied on subcircuit papers, for this to work the circuit itself should be aware of the fixed mode.

Do you have some thoughts on this?

Fixes #4.

tilk commented:
I'll take a second look at the CSS classes later, for now: there is a console.log in the code.

tilk commented on src/index.js:
```diff
@@ -149,6 +149,11 @@ export class Circuit extends HeadlessCircuit {
             });
         });
+        paper.fixed = function(fixed) {
+            console.log(this, fixed);
```
Please remove this debug statement.

ialokim commented:
> there is a console.log in the code.

True, good catch. Anyway, this is not yet ready to be merged, I just wanted to open an "work in progress" PR to get your feedback.

Would you prefer the subcircuits to be fixed at the same time (i.e. the method should `fixed` should be moved to the circuit class) or would you expect the website using the library to call the `fixed` function on every open paper?

tilk commented:
I think that having this configurable per paper is OK. I wonder about having additional function for setting this globally (for all open papers, maybe also for new papers?), as this would require tracking open papers in some way (both creation and removal would need to be tracked). I don't have a strong opinion on the matter. If the tracking can be done in a simple and error-free way, it's probably OK. But leaving this to the application is also fine.

ialokim commented:
I've now reworked and simplified the css rules and added proper handling for subcircuits as a client-side example inside `template.html`.

I think this should be ready to be merged now.

tilk commented on src/index.js:
```diff
@@ -72,6 +72,7 @@ export class Circuit extends HeadlessCircuit {
         return this.makePaper(elem, this._graph);
     }
     makePaper(elem, graph, opts) {
+        var circuit = this;
```
This should be a const, as it's not supposed to be changed.
```suggestion
        const circuit = this;
```
